rules/library.bzl

```diff
@@ -533,6 +515,20 @@ def apple_library(name, library_tools = {}, export_private_headers = True, names
             **kwargs
         )
         lib_names += [swift_libname]
+
+        # Add generated swift header to header maps for double quote imports
+        swift_doublequote_hmap_name = name + "_swift_doublequote_hmap"
+        headermap(
+            name = swift_doublequote_hmap_name,
+            namespace = namespace,
+            hdrs = [],
+            direct_hdr_providers = [swift_libname],
```

Comment:
Echoing @ob's question from above, I assume what we're doing here is passing an instantiated `swift_library` rule into the `headermap` rule.  In my beginner Bazel adopters mind, what's happening here isn't that we are passing some artifact, we are passing the information that the instantiated `swift_library` rules wants its consumers it to have: in this case, `swift_library` generates a `-Swift.h` header.  In Bazel terms, the information an instantiated rule wants its consumers to be able to access is called a `Provider`, and `Provider`s can be thought of as the "return value" of the instantiating a rule.

In this case, a `headermap` rule would like to include information about the instantiated `swift_library`s -Swift.h`header into a header map, so that ObjC classes in targets that consume this `apple_library`  can call methods on the Swift classes that the `swift_library` exposes in its header.

Is that more or less what's happening here?

 so here we pass the instantiated `swift_libname` (a provider) into `direct_hdr_providers` 

Inside the `headermap` rule's implementation, the `direct_hdr_providers` list  a `Provider` (@ob asked if was an ObjC provider - I don't have any intuition for what it is).  My assumption is that

Reply:
That is exactly what’s going on.

Reply:
Awesome.  Thanks for confirming @ob!